python/dask_cudf/dask_cudf/sorting.py

```diff
@@ -176,13 +176,14 @@ def rearrange_by_hash(
             for inp in inputs
         }
 
+        k_list = list(range(k))
```

Comment:
Why are we materializing a list here instead of just iterating via the range?

Reply:
Connected offline, it turns out that iterating over a range causes a materialization of a Python integer at each iteration which is somewhat expensive, whereas turning this into a list materializes it only once.